src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -1144,3 +1144,27 @@ void WorldSession::HandleCloseInteraction(WorldPackets::Misc::CloseInteraction&
     if (_player->PlayerTalkClass->GetInteractionData().SourceGuid == closeInteraction.SourceGuid)
         _player->PlayerTalkClass->GetInteractionData().Reset();
 }
+
+void WorldSession::HandleAdventureJournalOpenQuest(WorldPackets::Misc::AdventureJournalOpenQuest& packet)
+{
+    auto adventureJournalEntry = sAdventureJournalStore[packet.AdventureJournalID];
+    if (!adventureJournalEntry)
+        return;
+
+    Quest const* quest = sObjectMgr->GetQuestTemplate(adventureJournalEntry->QuestID);
+    if (!quest)
+        return;
+
+    if (_player->HasQuest(adventureJournalEntry->QuestID) && _player->CanTakeQuest(quest, true))
```

Comment:
```suggestion
    if (_player->hasQuest(adventureJournalEntry->QuestID) && _player->CanTakeQuest(quest, true))
```
My dumbass accidentally capitalized this